react-context/flow.txt

```diff
@@ -0,0 +1,69 @@
+User can sign up
+- When the app loads, the page gets rendered with the state isAuthorizing:true
+  and user:null
+- Once componentDidMount() runs for the first time, this.state.isAuthorizing is
+  set to false which renders the page with the Auth React component where the
+  user has the option to either sign-in or sign-up
+- If the path chosen is to sign-up, the sign-up page is rendered
+- Once user provides a new username and password then click 'Register', client's
+  POST request is made to api/auth/sign-up where:
+    - the server first checks to see if the request body contains both username
+      and password (along with its values)
+          - If not, a ClientError is thrown
+          - If both username and password are provided:
+            - the login info is stringified and saved in this.state
```

Comment:
this part is out of place, as the request has already been sent to the server.

Reply:
line 14 moved to be right after "Once user provides a new username and password, then click 'Register'"